### PR TITLE
michabo: init at 0.1

### PR DESCRIPTION
**Pull request by fgaz**
###### Motivation for this change

A native pleroma/mastodon client: https://git.pleroma.social/kaniini/michabo

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

There's no application icon, as adding it breaks all the in-app icons, as those need to be properly installed too since they use the same filename prefix. I might add it later if it does not get fixed upstream

### CODE REVIEW

File: pkgs/applications/misc/michabo/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  nativeBuildInputs = [
+    qmake
+  ];
+  buildInputs = [
+    qtbase
+    qtwebsockets
+  ];
+
+  configurePhase = "qmake michabo.pro DESTDIR=$out/bin";
```
Comment (jtojnar) [inline pkgs/applications/misc/michabo/default.nix]: What about using

```nix
  qmakeFlags = [ "michabo.pro" "DESTDIR=${placeholder "out"}/bin" ];
```

Though it looks like the shipped `.pro` file is incorrect, as it ~~ignores the `PREFIX` set in the [setup hook](https://github.com/NixOS/nixpkgs/blob/a4860c9cb2407db392615399fda54eaca7a8c71c/pkgs/development/libraries/qt-5/hooks/qmake-hook.sh#L4).~~ installs the program during build?

Comment (fgaz) [reply pkgs/applications/misc/michabo/default.nix]: Nice! didn't know about qmakeFlags and placeholder, thanks for the suggestion. I updated the commit.

I'll report the .pro problem upstream.

**Comment by fgaz** [comment]
Pointed out the installation during build [here](https://git.pleroma.social/kaniini/michabo/issues/3#note_37587).

Is it a blocker for merging this? Anything else?

File: pkgs/applications/misc/michabo/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
+{ stdenv
```
Comment (aanderse) [inline pkgs/applications/misc/michabo/default.nix]: -> `lib`

File: pkgs/applications/misc/michabo/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+}:
+
+let
+  desktopFile = writeText "michabo.desktop" ''
+    [Desktop Entry]
+    Type=Application
+    Name=Michabo
+    Exec=Michabo
+  '';
+in stdenv.mkDerivation rec {
```
Comment (aanderse) [inline pkgs/applications/misc/michabo/default.nix]: @fgaz please see https://github.com/NixOS/nixpkgs/issues/65399.

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/michabo/default.nix (hunk 3 above)
Comment (fgaz) [reply pkgs/applications/misc/michabo/default.nix]: Thanks, fixed

File: pkgs/applications/misc/michabo/default.nix [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+, writeText
+, fetchFromGitLab
+, qmake
+# qt
+, qtbase
+, qtwebsockets
+}:
+
+let
+  desktopFile = writeText "michabo.desktop" ''
```
Comment (aanderse) [inline pkgs/applications/misc/michabo/default.nix]: @fgaz this slipped by me first time around. Any reason not to use `makeDesktopItem` here instead?

**Review by aanderse — CHANGES_REQUESTED** [review]

File: pkgs/applications/misc/michabo/default.nix (hunk 4 above)
Comment (fgaz) [reply pkgs/applications/misc/michabo/default.nix]: None other than me not knowing about it :)
Updated

**Comment by aanderse** [comment]
@petabyteboy @ashkitten @alyssais as users of `mastodon` do any of you have interest in giving this a quick test?

**Comment by petabyteboy** [comment]
I can confirm it works :+1: 
screenshot

**Review by aanderse — APPROVED** [review]
Thanks @fgaz and @petabyteboy! :tada:

**Pull request merged**